build-tools/packages/build-cli/src/library/repoPolicyCheck/fluidBuildTasks.ts

```diff
@@ -345,16 +345,12 @@ function hasTaskDependency(
 	const packageSpecificSearchDeps = searchDeps.filter((d) => d.includes("#"));
 	/**
 	 * Set of package dependencies
-	 * True dependencies would include peer dependencies as well. Here we are
-	 * matching {@link Package.combinedDependencies} that is used in
-	 * {@link @fluidframework/build-tools/src/fluidBuild/buildGraph.ts#BuildGraph.initializePackages}
-	 * Being more conservative is not terrible as it would just look like a miss
-	 * from ^ specification and could be fixed with an explicit dependency.
 	 */
 	const packageDependencies = new Set([
 		...Object.keys(json.dependencies ?? {}),
 		// devDeps are not regular task deps, but might happen for internal type only packages.
 		...Object.keys(json.devDependencies ?? {}),
+		...Object.keys(json.peerDependencies ?? {}),
```

Comment:
I think this may cause fluid-build to fail if any peer deps are missing in the deps tree. It does some checking of node_modules to be sure everything is present, and it doesn't take into account any peerDep ignores we may have in the root package.json

Reply:
I believe you are referring to this code in build-tools/packages/build-tools/src/fluidBuild/symlinkUtils.ts:
```typescript
	for (const { name: dep, version } of pkg.combinedDependencies) {
		const depBuildPackage = buildPackages.get(dep);
		// Check and fix link if it is a known package and version satisfy the version.
		// TODO: check of extraneous symlinks
		if (depBuildPackage) {
			const sameMonoRepo = MonoRepo.isSame(pkg.monoRepo, depBuildPackage.monoRepo);
```
While there is a dearth of comments as to what things actually represent in build-tools, I think `buildPackages` limits itself to in repo packages. I would want peers checked here as well generally. We can add exception support if ever needed - currently none is needed from my testing using build:fast in client.